### PR TITLE
feat: admin email verification donation scheduling

### PR DESCRIPTION
**Pull request by jessiiepeng**
## Implementation description. How did you make this change?
* added new schedulingService `sendAdminEmailVerificationAfterDonationScheduled`
* hardcoded test admin email to test
* styling is consistent with donor email 
Screen Shot 2022-02-04 at 1 42 30 PM

Screen Shot 2022-02-04 at 1 42 38 PM

screenshot of mobile view

## Steps to test
1. change hardcoded admin email within `sendAdminEmailVerificationAfterDonationScheduled` in `schedulingService.ts` to your test email
2. make donations and check if email is sent / and formatting is correct

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages follow conventional commits and are descriptive. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s) 
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR
- [ ] The appropriate tests if necessary have been written

### CODE REVIEW

File: backend/typescript/services/implementations/schedulingService.ts [hunk 1]
```diff
@@ -522,6 +665,12 @@ class SchedulingService implements ISchedulingService {
       retNewSchedule,
     );
 
+    // send email to admin
```
Comment (hanlinc27) [inline backend/typescript/services/implementations/schedulingService.ts:668]: nit:
```suggestion
```

Comment (hanlinc27) [reply backend/typescript/services/implementations/schedulingService.ts:668]: with a descriptive name for the function, I think we can take out this comment :)

File: backend/typescript/services/implementations/schedulingService.ts [hunk 2]
```diff
@@ -321,6 +321,149 @@ class SchedulingService implements ISchedulingService {
...
+      const errorMessage =
+        "Attempted to call sendAdminEmailVerificationAfterDonationScheduled but this instance of SchedulingService does not have an EmailService instance";
+      Logger.error(errorMessage);
+      throw new Error(errorMessage);
+    }
+
+    try {
+      const { firstName, lastName } = donor;
+      // HARD CODED TEST ADMIN EMAIL
+      const adminEmail = "jessiepeng@uwblueprint.org";
```
Comment (hanlinc27) [inline backend/typescript/services/implementations/schedulingService.ts:338]: can you create a ticket for us to remember to change it back? 

I would also lift this to be a static constant for the whole class, something like:
`static TEMP_ADMIN_EMAIL = "jessiepeng@uwblueprint.org"`

File: backend/typescript/services/implementations/schedulingService.ts [hunk 3]
```diff
@@ -321,6 +321,149 @@ class SchedulingService implements ISchedulingService {
...
+
+      let volunteerTime = "";
+
+      if (schedule.volunteerTime) {
+        const timePartArray = schedule.volunteerTime.split(":");
+        const AmOrPm = parseInt(timePartArray[0], 10) >= 12 ? "PM" : "AM";
+        const hour = parseInt(timePartArray[0], 10) % 12 || 12;
+        timePartArray[0] = String(hour);
+        volunteerTime = `${timePartArray[0]}:${timePartArray[1]} ${AmOrPm}`;
+      }
```
Comment (hanlinc27) [inline backend/typescript/services/implementations/schedulingService.ts:384]: see the comment I made here: https://github.com/uwblueprint/community-fridge-kw/pull/122#pullrequestreview-873903550

File: backend/typescript/services/implementations/schedulingService.ts [hunk 4]
```diff
@@ -321,6 +321,149 @@ class SchedulingService implements ISchedulingService {
...
+            "dddd",
+          )}`;
+        } else {
+          frequencyString = `Monthly on the ${ordinal(
+            Number(dayjs(startTimeToLocalDate).format("D")),
+          )}`;
+        }
+      }
+
+      let volunteerTime = "";
```
Comment (hanlinc27) [inline backend/typescript/services/implementations/schedulingService.ts:376]: this is the _exact same code_ as our donor scheduling verification, 
can we please move this into either a utility function or a separate function in this class that accepts `startTime, endTime` as input params and returns all the formatted variables you need?

Comment (hanlinc27) [reply backend/typescript/services/implementations/schedulingService.ts:376]: then call on this function for both the donor email and the admin email

**Review by hanlinc27 — CHANGES_REQUESTED** [review]
I didn't realize the copy is almost the exact same barring one line for the emails, 
After this is merged https://github.com/uwblueprint/community-fridge-kw/pull/122 I think it's better if we have just **one function** for `sendEmailVerificationAfterSchedulingADonation` and then have it accept a `isAdmin` input parameter

You can then format the first line to either be "Thank you for scheduling a donation to your local community fridge" or "X has scheduled a donation" depending on the parameter

**Comment by jessiiepeng** [comment]
> I didn't realize the copy is almost the exact same barring one line for the emails, After this is merged #122 I think it's better if we have just **one function** for `sendEmailVerificationAfterSchedulingADonation` and then have it accept a `isAdmin` input parameter
> 
> You can then format the first line to either be "Thank you for scheduling a donation to your local community fridge" or "X has scheduled a donation" depending on the parameter and also the email it's being sent to depending on the parameter

Just made the changes and merged everything into one function! Made `isAdmin: Boolean` and it works fine! Also created the ticket to remind us to change the `TEMP_ADMIN_EMAIL`. Merged from the other email styling branch as well.

**Review by hanlinc27 — APPROVED** [review]
thanks for addressing the changes!

**Pull request merged**